src/System.Threading/tests/MonitorTests.cs

```diff
@@ -0,0 +1,145 @@
+            Assert.Throws<SynchronizationLockException>(() => Monitor.Exit(new object()));
+            Assert.Throws<SynchronizationLockException>(() => Monitor.Exit(valueType));
+        }
+
+        [Fact]
+        public static void IsEntered_Invalid()
+        {
+            var obj = new object();
+            Assert.Throws<ArgumentNullException>(() => Monitor.IsEntered(null));
+        }
+
+        [Fact]
+        public static void Pulse_Invalid()
+        {
+            var obj = new object();
+            Assert.Throws<ArgumentNullException>(() => Monitor.Pulse(null));
+        }
+
+        [Fact]
+        public static void PulseAll_Invalid()
+        {
+            var obj = new object();
+            Assert.Throws<ArgumentNullException>(() => Monitor.PulseAll(null));
+        }
+
+        [Fact]
+        public static void TryEnter_Invalid()
+        {
+            bool lockTaken = false;
+            var obj = new object();
+
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null, ref lockTaken));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null, 1));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null, 1, ref lockTaken));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null, TimeSpan.Zero));
+            Assert.Throws<ArgumentNullException>(() => Monitor.TryEnter(null, TimeSpan.Zero, ref lockTaken));
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => Monitor.TryEnter(null, -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => Monitor.TryEnter(null, -1, ref lockTaken));
+            Assert.Throws<ArgumentOutOfRangeException>(() => Monitor.TryEnter(null, TimeSpan.FromMilliseconds(-1)));
+            Assert.Throws<ArgumentOutOfRangeException>(() => Monitor.TryEnter(null, TimeSpan.FromMilliseconds(-1), ref lockTaken));
+
+            lockTaken = true;
+            Assert.Throws<ArgumentException>(() => Monitor.TryEnter(obj, ref lockTaken));
+            lockTaken = true;
+            Assert.Throws<ArgumentException>(() => Monitor.TryEnter(obj, 0, ref lockTaken));
+            lockTaken = true;
+            Assert.Throws<ArgumentException>(() => Monitor.TryEnter(obj, TimeSpan.Zero, ref lockTaken));
+        }
+
+        [Fact]
+        public static void Wait_Invalid()
+        {
+            var obj = new object();
+            Assert.Throws<ArgumentNullException>(() => Monitor.Wait(null));
+            Assert.Throws<ArgumentNullException>(() => Monitor.Wait(null, 1));
+            Assert.Throws<ArgumentNullException>(() => Monitor.Wait(null, TimeSpan.Zero));
+            Assert.Throws<ArgumentOutOfRangeException>(() => Monitor.Wait(null, -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => Monitor.Wait(null, TimeSpan.FromMilliseconds(-1)));
+        }
+    }
+}
```

Comment:
I didn't have any specific plans to add more tests, but we really should add some more real-world scenarios beyond simple validation. Something along the lines of what we do in the Semaphore tests, for example.